common/content-api.js

```diff
@@ -8,59 +10,37 @@ const map = require('lodash/fp/map');
 const pick = require('lodash/pick');
 const sortBy = require('lodash/fp/sortBy');
 
-const request = require('request-promise-native');
-const querystring = require('querystring');
-
 const logger = require('./logger');
+const { sanitiseUrlPath } = require('./urls');
+const { CONTENT_API_URL } = require('./secrets');
 
 const getAttrs = response => get('data.attributes')(response);
 const mapAttrs = response => map('attributes')(response.data);
 
-const { sanitiseUrlPath } = require('./urls');
-let { CONTENT_API_URL } = require('./secrets');
-
-function fetch(urlPath, options) {
-    logger.debug(
-        `Fetching ${CONTENT_API_URL}${urlPath}${
-            options && options.qs ? '?' + querystring.stringify(options.qs) : ''
-        }`
-    );
-
-    const defaults = {
-        url: `${CONTENT_API_URL}${urlPath}`,
-        json: true
-    };
-    const params = Object.assign({}, defaults, options);
-    return request(params);
-}
-
-/**
- * Fetch all locales for a given url path
- * Usage:
- * ```
- * fetchAllLocales(reqLocale => {
- *   return `/v1/${reqLocale}/funding-programmes`
- * }).then(responses => ...)
- * ```
- */
-function fetchAllLocales(toUrlPathFn, options = {}) {
-    const urlPaths = ['en', 'cy'].map(toUrlPathFn);
-    const promises = urlPaths.map(urlPath => fetch(urlPath, options));
-    return Promise.all(promises);
-}
+const queryContentApi = got.extend({
+    prefixUrl: CONTENT_API_URL,
+    headers: { 'user-agent': 'tnlcf-www' },
+    hooks: {
+        beforeRequest: [
+            function(options) {
+                logger.debug(`Fetching ${options.url.href}`);
```

Comment:
Does this end up in DataDog? It would massively inflate our usage (eg. we pay more... not a lot more, but still!) if it logs for essentially every webpage request (multiple times in some cases). Or do the `debug` requests only log locally? Either way, would be nice to get the queryParams in here too as they sometimes form the bulk of the useful info (eg. calls to the listing endpoint).

Reply:
debug only logs locally and `options.url.href` includes the query params. This is essentially a 1-to-1 replacement for what was already there.